sklearn/metrics/_plot/roc_curve.py

```diff
@@ -46,25 +51,28 @@ def plot(self, ax=None, name=None, **kwargs):
 
         Parameters
         ----------
-        ax : Matplotlib Axes or None, default=None
-            Axes object to plot on.
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
 
-        name : str or None, default=None
+        name : str, default=None
             Name of ROC Curve for labeling. If `None`, use the name of the
             estimator.
         """
-        check_matplotlib_support('plot_roc_curve')
+        check_matplotlib_support('RocCurveDisplay.plot')
         import matplotlib.pyplot as plt
 
         if ax is None:
             fig, ax = plt.subplots()
 
         name = self.estimator_name if name is None else name
 
-        if 'label' not in kwargs:
-            label = "{} (AUC = {:0.2f})".format(name, self.roc_auc)
-            kwargs['label'] = label
-        self.line_ = ax.plot(self.fpr, self.tpr, **kwargs)[0]
+        label_kwargs = {
```

Comment:
is this really nicer? Now you're using something called ``label_kwargs`` to store all ``plot`` kwargs. That's confusing to me.

Reply:
Ah this should be `line_kwargs` not `label_kwargs`,